copy/copy.go

```diff
@@ -600,13 +580,31 @@ func (ic *imageCopier) copyUpdatedConfigAndManifest(ctx context.Context) ([]byte
 		return nil, err
 	}
 
+	ic.c.progressPool.Wait()
```

Comment:
The documentation of `Progress.Wait` says
> After this method has been called, there is no way to reuse *Progress instance.

The middle of `copyUpdatedConfigAndManifest` is not a good place for such a call, because `copyUpdatedConfigAndManifest` can be called multiple times per image (as well, as, strictly speaking, zero times if an error happens before).  Either the pool should live across all of those calls (and in that case the “Writing manifest to image destination” message just below needs to happen in some way that does not interfere with the `progressPool`), or it should be closed before even starting to deal with configs/manifests (using just a message, not progress bar, for the config, or maybe creating an one-use single-bar progress pool just for each attempt to copy a manifest).

(Similar in nature is a concern about attaching `progressPool` to `copier` vs. `imageCopier`; right now that makes ~zero difference; longer-term after #400 is resurrected, `copier` might live across copies of multiple single-arch images, whereas `imageCopier` would carry state related to one of them.)

(Purely aesthethically it would be nice for the “closing” of the progress pool to happen (or, to be guaranteed to happen as a last-resort backstop) at the same level of the call stack that creates the progress pool.)

Reply:
Again, that's intended. We need to make sure the progress bar has finished rendering before printing, see the next line `ic.c.Printf("Writing manifest to image destination\n")`. I understand it's not ideal when looking at the assumptions on the code, but I could not find any other way and am very hesitant to rewrite `copy.go`.

Reply:
> (Similar in nature is a concern about attaching `progressPool` to `copier` vs. `imageCopier`; right now that makes ~zero difference; longer-term after #400 is resurrected, `copier` might live across copies of multiple single-arch images, whereas `imageCopier` would carry state related to one of them.)

It's possible to add new bars dynamically and it will render correctly as long as nothing prints in between.

Reply:
> (Purely aesthethically it would be nice for the “closing” of the progress pool to happen (or, to be guaranteed to happen as a last-resort backstop) at the same level of the call stack that creates the progress pool.)

I share that desire but mpb doesn't expose a `Nuke()` API. It is possible though to nuke each bar indivually which would allow to work around that.